presidio-analyzer/analyzer/predefined_recognizers/sg_fin_recognizer.py

```diff
@@ -0,0 +1,23 @@
+from analyzer import Pattern
+from analyzer import PatternRecognizer
+
+# pylint: disable=line-too-long,abstract-method
+# Weak pattern: all FIN number start with "S", "T", "F" or "G"
+# and ends with a character, e.g., G3311100L
+# Ref: https://en.wikipedia.org/wiki/National_Registration_Identity_Card
+WEAK_REGEX = r'(\b[a-z,A-Z][0-9]{7}[a-z,A-Z]\b)'
+MEDIUM_REGEX = r'(\b[s,t,f,g][0-9]{7}[a-z,A-Z]\b)'
```

Comment:
Maybe something like this makes more sense : [STFGstfg]. This will ensure both cases are being checked for a strong match.

Reply:
good point @rahulraina7 updated with your suggestion